internal/util/update/update_test.go

```diff
@@ -272,104 +273,183 @@ func TestCommand_Run_noAdd(t *testing.T) {
+		"update using resource-merge strategy with local changes": {
+			strategy: KResourceMerge,
+			initialUpstream: testutil.Content{
+				Data:   testutil.Dataset1,
+				Branch: "master",
+			},
+			updatedUpstream: testutil.Content{
+				Data: testutil.Dataset2,
+			},
+			updatedLocal: testutil.Content{
+				Data: testutil.Dataset3,
+			},
+			expectedLocal: testutil.Content{
+				Data: testutil.DatasetMerged,
+			},
+			expectedCommit: func(writer *testutil.TestSetupManager) string {
+				c, _ := writer.UpstreamRepo.GetCommit()
+				return c
+			},
+		},
+		"update using fast-forward strategy with local changes": {
+			strategy: FastForward,
+			initialUpstream: testutil.Content{
+				Data:   testutil.Dataset1,
+				Branch: "master",
+			},
+			updatedUpstream: testutil.Content{
+				Data: testutil.Dataset2,
+			},
+			updatedLocal: testutil.Content{
+				Data: testutil.Dataset3,
+			},
+			expectedLocal: testutil.Content{
+				Data: testutil.Dataset3,
+			},
+			expectedErr: "local package files have been modified",
+			expectedCommit: func(writer *testutil.TestSetupManager) string { // expect Kptfile to keep the commit
 				f, err := kptfileutil.ReadFile(filepath.Join(writer.LocalWorkspace.WorkspaceDirectory, writer.UpstreamRepo.RepoName))
 				if !assert.NoError(writer.T, err) {
 					return ""
 				}
 				return f.Upstream.Git.Commit
 			},
 		},
-		// // forcedeletereplace should reset hard to dataset 2 -- upstream modified copy
-		// {ForceDeleteReplace,
-		//	 testutil.Dataset2, // expect the upstream changes
-		//	 "",                // expect no error
-		//	 func(writer *testutil.TestSetupManager) string {
-		// 		c, _ := writer.UpstreamRepo.GetCommit() // expect the upstream commit
-		// 		return c
-		//	 },
-		// },
-		// // gitpatch should create a merge conflict between 2 and 3
-		// {AlphaGitPatch,
-		// 	 testutil.UpdateMergeConflict,     // expect a merge conflict
-		//	 "Failed to merge in the changes", // expect an error
-		//	 func(writer *testutil.TestSetupManager) string {
-		// 		c, _ := writer.UpstreamRepo.GetCommit() // expect the upstream commit as a staged change
-		// 		return c
-		// 	 },
-		// },
-		{KResourceMerge,
-			testutil.DatasetMerged, // expect a merge conflict
-			"",                     // expect an error
-			func(writer *testutil.TestSetupManager) string {
-				c, _ := writer.UpstreamRepo.GetCommit() // expect the upstream commit as a staged change
+		"update using force-delete-replace strategy with local changes": {
+			strategy: ForceDeleteReplace,
+			initialUpstream: testutil.Content{
+				Data:   testutil.Dataset1,
+				Branch: "master",
+			},
+			updatedUpstream: testutil.Content{
+				Data: testutil.Dataset2,
+			},
+			updatedLocal: testutil.Content{
+				Data: testutil.Dataset3,
+			},
+			expectedLocal: testutil.Content{
+				Data: testutil.Dataset2,
+			},
+			expectedCommit: func(writer *testutil.TestSetupManager) string {
+				c, _ := writer.UpstreamRepo.GetCommit()
```

Comment:
Does this return an error ? Shouldn't we assert noError here ?

Reply:
Updated.